packages/ui/components/form/select/Select.tsx

```diff
@@ -1,109 +1,96 @@
 import { useId } from "@radix-ui/react-id";
 import * as React from "react";
-import type {
-  GroupBase,
-  Props,
-  SingleValue,
-  MultiValue,
-  SelectComponentsConfig,
-  MenuPlacement,
-} from "react-select";
-import ReactSelect, { components as reactSelectComponents } from "react-select";
+import type { GroupBase, Props, SingleValue, MultiValue } from "react-select";
+import ReactSelect from "react-select";
 
 import classNames from "@calcom/lib/classNames";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 
 import { Label } from "../inputs/Label";
-import {
-  ControlComponent,
-  InputComponent,
-  MenuComponent,
-  MenuListComponent,
-  OptionComponent,
-  SingleValueComponent,
-  ValueContainerComponent,
-  MultiValueComponent,
-} from "./components";
+import { getReactSelectProps } from "./selectTheme";
 
 export type SelectProps<
   Option,
   IsMulti extends boolean = false,
   Group extends GroupBase<Option> = GroupBase<Option>
 > = Props<Option, IsMulti, Group>;
 
-export const getReactSelectProps = <
-  Option,
-  IsMulti extends boolean = false,
-  Group extends GroupBase<Option> = GroupBase<Option>
->({
-  className,
-  components,
-  menuPlacement = "auto",
-}: {
-  className?: string;
-  components: SelectComponentsConfig<Option, IsMulti, Group>;
-  menuPlacement?: MenuPlacement;
-}) => ({
-  menuPlacement,
-  className: classNames("block min-h-6 w-full min-w-0 flex-1 rounded-md", className),
-  classNamePrefix: "cal-react-select",
-  components: {
-    ...reactSelectComponents,
-    IndicatorSeparator: () => null,
-    Input: InputComponent,
-    Option: OptionComponent,
-    Control: ControlComponent,
-    SingleValue: SingleValueComponent,
-    Menu: MenuComponent,
-    MenuList: MenuListComponent,
-    ValueContainer: ValueContainerComponent,
-    MultiValue: MultiValueComponent,
-    ...components,
-  },
-});
-
 export const Select = <
   Option,
   IsMulti extends boolean = false,
   Group extends GroupBase<Option> = GroupBase<Option>
 >({
-  className,
   components,
-  styles,
+  menuPlacement,
   ...props
 }: SelectProps<Option, IsMulti, Group>) => {
   const reactSelectProps = React.useMemo(() => {
     return getReactSelectProps<Option, IsMulti, Group>({
-      className,
       components: components || {},
+      menuPlacement,
     });
-  }, [className, components]);
+  }, [components, menuPlacement]);
 
+  // Annoyingly if we update styles here we have to update timezone select too
```

Comment:
Could you explain a bit more in this PR why that's not possible? Would be really great if we could indeed use some sort of generator function so we can't forget to update the timezone select 😁